service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/GetFederatedIdentityStep.java

```diff
@@ -88,7 +91,7 @@ private boolean federatedIdentityExists(
 
   private boolean k8sServiceAccountExists(String uamiName, CoreV1Api aksApi) {
     try {
-      return aksApi.readNamespacedServiceAccount(uamiName, federatedCredentialName, null) != null;
+      return aksApi.readNamespacedServiceAccount(uamiName, k8sServiceAccountName, null) != null;
```

Comment:
This was the ultimate source of the bug. The old code assumed that the name of the KSA was the same as the the federated credentials. This is no longer true. The meant that the overall result of `GetFederatedIdentityStep` was that the identity did not exist so `DeleteFederatedIdentityStep` did a noop. In the end I decided to skip the check for the delete flight anyway since we want to delete the fed creds even if the KSA does not exist.

Reply:
But there were 2 bugs, correct? This one and the one mentioned above-- https://github.com/DataBiosphere/terra-workspace-manager/pull/1488/files#r1337672353